hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.consumer;
+
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+
+/**
+ *  Beta API subject to change.
+ * 
+ * @author Tim Taylor {@literal<timt@netflix.com>}
+ */
+public class HollowConsumer {
+
+    // TODO: timt: is this needed, or do we just use a HollowReadStateEngine in place of this? Created for now
+    //   to have symmetry with HollowProducer.WriteState
+    public static interface ReadState {
```

Comment:
Producer restore and integrity check code for loading read states is still looking for a good home in the Consumer side of the API. Jury is still out on whether `ReadState` justify's its own weight as compared to passing a `HollowReadStateEngine`. It does provide a lighter weight way to couple a read state engine with the version ID used to arrive at it (lighter weight than passing a `HollowClient`. However, a read state engine could be mutated under the covers making a `ReadState` tell a lie.